src/client/stdlib/duckdb.js

```diff
@@ -255,7 +255,8 @@ async function insertFile(database, name, file, options) {
         if (/\.parquet$/i.test(file.name)) {
           return await connection.query(`CREATE VIEW '${name}' AS SELECT * FROM parquet_scan('${file.name}')`);
         }
-        throw new Error(`unknown file type: ${file.mimeType}`);
+        // Lastly, let DuckDB guess: it may be a .duckdb (or .db or .ddb) database file.
+        return await connection.query(`ATTACH '${file.name}' AS ${name} (READ_ONLY)`);
```

Comment:
I changed my mind here and would like to be more conservative and only do this with `.db`, `.duckdb`, `.ddb`. It’ll make it easier for us to introduce new formats in the future, and encourage users to be more consistent with how they store DuckDB database files.

Reply:
(But I think we should only check the file extension; we don’t need to check the additional magic.)